### ISSUE TITLE
Request : force fullscreen in landscape on mobile devices

### ISSUE THREAD

**Issue by r4dius** [Selz/plyr#584]
- [x] Issue does not already exist
- [ ] Issue observed on https://plyr.io

### Expected behaviour 

Not sure if it should be a default behaviour but when using the chrome player fullscreen button we get the video forced in landscape mode which is nice, same thing with the youtube app, even if you don't enable landscape or auto rotation on the device, in plyr there's no way to force landscape fullscreen like that

- Browser: Chrome
- Version: 52
- Operating System: Android
- Version: 7.1.1

Players affected:
- [x] HTML5 Video
- [ ] HTML5 Audio
- [ ] YouTube
- [ ] Vimeo

### Steps to reproduce 

- Click fullscreen when device is in portrait

**Comment by sampotts** [comment]
Good idea. Will look into it... 👍

**Comment by ammein** [comment]
Yes. I've been looking solutions for that Mobile Orientation. By default behaviour using <video src="somelink" **allowfullscreen** > that allowfullscreen works perfectly when pressed for fullscreen, automatically landscape orientation for mobile. We need this feature ASAP ! . Also, I've been thinking using eventlistener for **orientationchange** where a device is using mobile size.

**Comment by sampotts** [comment]
@ammein Get a pull request in 👍

**Comment by Wanderish** [comment]
@ammein I tried doing that but in vain. Any other suggestions?

**Comment by Reqrefusion** [comment]
I am wait for next version

**Comment by r4dius** [comment]
Okay, so after forgetting about this just found an easy way to get this working, I'm using some code on my custom javascript not tested to add to plyr yet,
When fullscreen toggling is done just force "landscape", this will align on the actual landscape orientation
:
$(document).on('webkitfullscreenchange mozfullscreenchange fullscreenchange MSFullscreenChange', function() {
	if(plyr.fullscreen.active) {
		screen.orientation.lock("landscape");
	} else {
		screen.orientation.unlock();
	}
}

Would need to be disabled if height > width

**Comment by sampotts** [comment]
Looks like there's no support in Safari:
https://developer.mozilla.org/en-US/docs/Web/API/ScreenOrientation/lock

But we can always add this in...

**Comment by nitaaikumar** [comment]
> But we can always add this in...

Just a reminder to please add this in so that Plyr can be on par with the normal html5 player in Android Chrome.

**Comment by CodeDruid13** [comment]
> Looks like there's no support in Safari:
> https://developer.mozilla.org/en-US/docs/Web/API/ScreenOrientation/lock
> 
> But we can always add this in...

Any progress yet?

**Comment by AradAral** [comment]
No updates on this one yet?!

**Comment by jawahar95** [comment]
No updates on this one yet?!

**Comment by sampotts** [comment]
No pull requests yet?!

**Comment by mert-ozkul** [comment]
`var winwidth=$(window).width();
            var winheight=$(window).height();

            const player = Plyr.setup('#player');
            player.forEach (item => {

                item.on('enterfullscreen', event => {

                 $(".plyr__video-wrapper").css({"transform": "rotate(90deg)", "height": winwidth, "width": winheight, "margin-left": "0px", "margin-right": "0px"});
                })
                item.on('exitfullscreen', event => {
                 $(".plyr__video-wrapper").css({"transform": "rotate(0deg)", "height": "100%", "width": "100%", "margin-left": "auto", "margin-right": "auto"});
                })
            });`

I also faced the same problem. And that's how I did it, it works very nicely!

**Comment by Bikoubikou** [comment]
I have same problem , you ca help me ?

<html lang="en" dir="ltr" data-cast-api-enabled="true">
    <head>
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <meta name="robots" content="noindex">
        <title>Plyr</title>
        <link rel="stylesheet" href="https://cdn.plyr.io/3.6.1/plyr.css">
        <style type="text/css" media="all">
            html,body.plyr{
                height:100%;
                width:100%;
                margin:0;
                overflow:
                hidden;
                background:#000
            }
            #player{
                width:100%;
                height:100%;
                overflow:hidden
            }
            .plyr--audio .plyr__control.plyr__tab-focus,
            .plyr--audio .plyr__control:hover,
            .plyr--audio .plyr__control[aria-expanded=true] {
                background: <?php echo $plyrcl; ?>!important
            }
            .plyr--video .plyr__control.plyr__tab-focus,
            .plyr--video .plyr__control:hover,
            .plyr--video .plyr__control[aria-expanded=true] {
                background: <?php echo $plyrcl; ?>!important
            }
            .plyr__control--overlaid:focus,
            .plyr__control--overlaid:hover {
                background: <?php echo $plyrcl; ?>!important
            }
            .plyr__menu__container .plyr__control[role=menuitemradio][aria-checked=true]::before {
                background: <?php echo $plyrcl; ?>!important
            }
            .plyr--full-ui input[type=range] {
                color: <?php echo $plyrcl; ?>!important
            }
            .plyr__control--overlaid {
                background: rgba(255, 255, 255, 0.5)!important
            }
            
       
        </style>
    </head>
    <body class="plyr">
        <video poster="<?php echo $prvimg; ?>" id="player" playsinline controls>
            <source src="<?php echo $mp4fle; ?>" type="video/mp4" size="720">
        </video>
    </body>
    <script src="https://cdn.plyr.io/3.6.1/plyr.polyfilled.js"></script>
    <script src="https://cdn.plyr.io/3.6.1/plyr.js" crossorigin="anonymous"></script>
    <script>const player = new Plyr('#player')</script>
    
    </html>
